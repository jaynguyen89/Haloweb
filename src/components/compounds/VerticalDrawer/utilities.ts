import { IconName } from '@fortawesome/free-regular-svg-icons';
import { IconDefinition } from '@fortawesome/free-solid-svg-icons';

export type TVerticalDrawerMenuItem = {
    isActive?: true,
    text: string,
    icon?: IconName | IconDefinition | string,
    link?: string,
    onClick?: () => void,
    subMenu?: Array<Omit<TVerticalDrawerMenuItem, 'subMenu'>>,
};

export type TVerticalDrawerMenu = {
    title: string,
    items: Array<TVerticalDrawerMenuItem>,
    divider?: true,
};

export const setActiveMenuItem = (
    menuItems: Array<TVerticalDrawerMenu>,
    keyword: string,
    searchBy: 'text' | 'link',
    inSubMenu: boolean = false,
): Array<TVerticalDrawerMenu> => {
    /* eslint-disable max-depth */
    for (const menuItem of menuItems)
        for (const item of menuItem.items)
            if (item.isActive) {
                item.isActive = undefined;

                if (item.hasOwnProperty('subMenu'))
                    for (const subMenuItem of item.subMenu!)
                        if (subMenuItem.isActive) subMenuItem.isActive = undefined;

                break;
            }
    /* eslint-enable max-depth */

    if (!inSubMenu) {
        for (const menuItem of menuItems)
            for (const item of menuItem.items) {
                if (searchBy === 'text' && item.text.includes(keyword))
                    item.isActive = true;

                if (searchBy === 'link' && item.link!.includes(keyword))
                    item.isActive = true;

                if (item.isActive) break;
            }

        return menuItems;
    }

    for (const menuItem of menuItems)
        for (const item of menuItem.items)
            if (item.hasOwnProperty('subMenu')) {
                for (const subMenuItem of item.subMenu!) {
                    if (searchBy === 'text' && subMenuItem.text.includes(keyword))
                        subMenuItem.isActive = true;

                    if (searchBy === 'link' && item.link!.includes(keyword))
                        subMenuItem.isActive = true;

                    if (subMenuItem.isActive) {
                        item.isActive = true;
                        break;
                    }
                }
            }

    return menuItems;
};
