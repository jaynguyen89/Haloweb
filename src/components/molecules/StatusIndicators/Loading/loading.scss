.stage {
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  padding: 32px 0;
  margin: 0 -5%;
  overflow: hidden;
  min-width: 50px;
}

/* CSS classes for Falling Dots */
/* ============================ */
.dot-falling {
  position: relative;
  left: -9999px;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  box-shadow: 9999px 0 0 0 #9880ff;
  animation: dot-falling 1s infinite linear;
  animation-delay: 0.1s;
}
.dot-falling::before, .dot-falling::after {
  content: "";
  display: inline-block;
  position: absolute;
  top: 0;
}
.dot-falling::before {
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  animation: dot-falling-before 1s infinite linear;
  animation-delay: 0s;
}
.dot-falling::after {
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  animation: dot-falling-after 1s infinite linear;
  animation-delay: 0.2s;
}

@keyframes dot-falling {
  0% {
    box-shadow: 9999px -15px 0 0 rgba(152, 128, 255, 0);
  }
  25%, 50%, 75% {
    box-shadow: 9999px 0 0 0 #9880ff;
  }
  100% {
    box-shadow: 9999px 15px 0 0 rgba(152, 128, 255, 0);
  }
}
@keyframes dot-falling-before {
  0% {
    box-shadow: 9984px -15px 0 0 rgba(152, 128, 255, 0);
  }
  25%, 50%, 75% {
    box-shadow: 9984px 0 0 0 #9880ff;
  }
  100% {
    box-shadow: 9984px 15px 0 0 rgba(152, 128, 255, 0);
  }
}
@keyframes dot-falling-after {
  0% {
    box-shadow: 10014px -15px 0 0 rgba(152, 128, 255, 0);
  }
  25%, 50%, 75% {
    box-shadow: 10014px 0 0 0 #9880ff;
  }
  100% {
    box-shadow: 10014px 15px 0 0 rgba(152, 128, 255, 0);
  }
}

/* CSS classes for Elastic Dots */
/* ============================ */
.dot-elastic {
  position: relative;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  animation: dot-elastic 1s infinite linear;
}
.dot-elastic::before, .dot-elastic::after {
  content: "";
  display: inline-block;
  position: absolute;
  top: 0;
}
.dot-elastic::before {
  left: -15px;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  animation: dot-elastic-before 1s infinite linear;
}
.dot-elastic::after {
  left: 15px;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  animation: dot-elastic-after 1s infinite linear;
}

@keyframes dot-elastic-before {
  0% {
    transform: scale(1, 1);
  }
  25% {
    transform: scale(1, 1.5);
  }
  50% {
    transform: scale(1, 0.67);
  }
  75% {
    transform: scale(1, 1);
  }
  100% {
    transform: scale(1, 1);
  }
}
@keyframes dot-elastic {
  0% {
    transform: scale(1, 1);
  }
  25% {
    transform: scale(1, 1);
  }
  50% {
    transform: scale(1, 1.5);
  }
  75% {
    transform: scale(1, 1);
  }
  100% {
    transform: scale(1, 1);
  }
}
@keyframes dot-elastic-after {
  0% {
    transform: scale(1, 1);
  }
  25% {
    transform: scale(1, 1);
  }
  50% {
    transform: scale(1, 0.67);
  }
  75% {
    transform: scale(1, 1.5);
  }
  100% {
    transform: scale(1, 1);
  }
}

/* CSS classes for Pulsing Dots */
/* ============================ */
.dot-pulse {
  position: relative;
  left: -9999px;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  box-shadow: 9999px 0 0 -5px;
  animation: dot-pulse 1.5s infinite linear;
  animation-delay: 0.25s;
}
.dot-pulse::before, .dot-pulse::after {
  content: "";
  display: inline-block;
  position: absolute;
  top: 0;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
}
.dot-pulse::before {
  box-shadow: 9984px 0 0 -5px;
  animation: dot-pulse-before 1.5s infinite linear;
  animation-delay: 0s;
}
.dot-pulse::after {
  box-shadow: 10014px 0 0 -5px;
  animation: dot-pulse-after 1.5s infinite linear;
  animation-delay: 0.5s;
}

@keyframes dot-pulse-before {
  0% {
    box-shadow: 9984px 0 0 -5px;
  }
  30% {
    box-shadow: 9984px 0 0 2px;
  }
  60%, 100% {
    box-shadow: 9984px 0 0 -5px;
  }
}
@keyframes dot-pulse {
  0% {
    box-shadow: 9999px 0 0 -5px;
  }
  30% {
    box-shadow: 9999px 0 0 2px;
  }
  60%, 100% {
    box-shadow: 9999px 0 0 -5px;
  }
}
@keyframes dot-pulse-after {
  0% {
    box-shadow: 10014px 0 0 -5px;
  }
  30% {
    box-shadow: 10014px 0 0 2px;
  }
  60%, 100% {
    box-shadow: 10014px 0 0 -5px;
  }
}

/* CSS classes for Flashing Dots */
/* ============================ */
.dot-flashing {
  position: relative;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  animation: dot-flashing 1s infinite linear alternate;
  animation-delay: 0.5s;
}
.dot-flashing::before, .dot-flashing::after {
  content: "";
  display: inline-block;
  position: absolute;
  top: 0;
}
.dot-flashing::before {
  left: -15px;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  animation: dot-flashing 1s infinite alternate;
  animation-delay: 0s;
}
.dot-flashing::after {
  left: 15px;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  animation: dot-flashing 1s infinite alternate;
  animation-delay: 1s;
}

@keyframes dot-flashing {
  0% {
    background-color: #9880ff;
  }
  50%, 100% {
    background-color: rgba(152, 128, 255, 0.2);
  }
}

/* CSS classes for Colliding Dots */
/* ============================ */

.dot-collision {
  position: relative;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
}
.dot-collision::before, .dot-collision::after {
  content: "";
  display: inline-block;
  position: absolute;
  top: 0;
}
.dot-collision::before {
  left: -10px;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  animation: dot-collision-before 2s infinite ease-in;
}
.dot-collision::after {
  left: 10px;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  animation: dot-collision-after 2s infinite ease-in;
  animation-delay: 1s;
}

@keyframes dot-collision-before {
  0%, 50%, 75%, 100% {
    transform: translateX(0);
  }
  25% {
    transform: translateX(-15px);
  }
}
@keyframes dot-collision-after {
  0%, 50%, 75%, 100% {
    transform: translateX(0);
  }
  25% {
    transform: translateX(15px);
  }
}

/* CSS classes for Rotating Dots */
/* ============================ */
.dot-carousel {
  position: relative;
  left: -9999px;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  box-shadow: 9984px 0 0 0 #9880ff, 9999px 0 0 0 #9880ff, 10014px 0 0 0 #9880ff;
  animation: dot-carousel 1.5s infinite linear;
}

@keyframes dot-carousel {
  0% {
    box-shadow: 9984px 0 0 -1px #9880ff, 9999px 0 0 1px #9880ff, 10014px 0 0 -1px #9880ff;
  }
  50% {
    box-shadow: 10014px 0 0 -1px #9880ff, 9984px 0 0 -1px #9880ff, 9999px 0 0 1px #9880ff;
  }
  100% {
    box-shadow: 9999px 0 0 1px #9880ff, 10014px 0 0 -1px #9880ff, 9984px 0 0 -1px #9880ff;
  }
}

/* CSS classes for Jumping Dots */
/* ============================ */
.dot-typing {
  position: relative;
  left: -9999px;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  box-shadow: 9984px 0 0 0 #9880ff, 9999px 0 0 0 #9880ff, 10014px 0 0 0 #9880ff;
  animation: dot-typing 1.5s infinite linear;
}

@keyframes dot-typing {
  0% {
    box-shadow: 9984px 0 0 0 #9880ff, 9999px 0 0 0 #9880ff, 10014px 0 0 0 #9880ff;
  }
  16.667% {
    box-shadow: 9984px -10px 0 0 #9880ff, 9999px 0 0 0 #9880ff, 10014px 0 0 0 #9880ff;
  }
  33.333% {
    box-shadow: 9984px 0 0 0 #9880ff, 9999px 0 0 0 #9880ff, 10014px 0 0 0 #9880ff;
  }
  50% {
    box-shadow: 9984px 0 0 0 #9880ff, 9999px -10px 0 0 #9880ff, 10014px 0 0 0 #9880ff;
  }
  66.667% {
    box-shadow: 9984px 0 0 0 #9880ff, 9999px 0 0 0 #9880ff, 10014px 0 0 0 #9880ff;
  }
  83.333% {
    box-shadow: 9984px 0 0 0 #9880ff, 9999px 0 0 0 #9880ff, 10014px -10px 0 0 #9880ff;
  }
  100% {
    box-shadow: 9984px 0 0 0 #9880ff, 9999px 0 0 0 #9880ff, 10014px 0 0 0 #9880ff;
  }
}

/* CSS classes for Circling Dots */
/* ============================ */
.dot-windmill {
  position: relative;
  top: -10px;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  transform-origin: 5px 15px;
  animation: dot-windmill 2s infinite linear;
}
.dot-windmill::before, .dot-windmill::after {
  content: "";
  display: inline-block;
  position: absolute;
}
.dot-windmill::before {
  left: -8.66254px;
  top: 15px;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
}
.dot-windmill::after {
  left: 8.66254px;
  top: 15px;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
}

@keyframes dot-windmill {
  0% {
    transform: rotateZ(0deg) translate3d(0, 0, 0);
  }
  100% {
    transform: rotateZ(720deg) translate3d(0, 0, 0);
  }
}

/* CSS classes for Floating Dots */
/* ============================ */
.dot-floating {
  position: relative;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  animation: dot-floating 3s infinite cubic-bezier(0.15, 0.6, 0.9, 0.1);
}
.dot-floating::before, .dot-floating::after {
  content: "";
  display: inline-block;
  position: absolute;
  top: 0;
}
.dot-floating::before {
  left: -12px;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  animation: dot-floating-before 3s infinite ease-in-out;
}
.dot-floating::after {
  left: -24px;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  animation: dot-floating-after 3s infinite cubic-bezier(0.4, 0, 1, 1);
}

@keyframes dot-floating {
  0% {
    left: calc(-50% - 5px);
  }
  75% {
    left: calc(50% + 105px);
  }
  100% {
    left: calc(50% + 105px);
  }
}
@keyframes dot-floating-before {
  0% {
    left: -50px;
  }
  50% {
    left: -12px;
  }
  75% {
    left: -50px;
  }
  100% {
    left: -50px;
  }
}
@keyframes dot-floating-after {
  0% {
    left: -100px;
  }
  50% {
    left: -24px;
  }
  75% {
    left: -100px;
  }
  100% {
    left: -100px;
  }
}

/* CSS classes for Spinning Dots */
/* ============================ */
.dot-spin {
  position: relative;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: transparent;
  color: transparent;
  box-shadow: 0 -18px 0 0 #9880ff, 12.727926px -12.727926px 0 0 #9880ff, 18px 0 0 0 #9880ff, 12.727926px 12.727926px 0 0 rgba(152, 128, 255, 0), 0 18px 0 0 rgba(152, 128, 255, 0), -12.727926px 12.727926px 0 0 rgba(152, 128, 255, 0), -18px 0 0 0 rgba(152, 128, 255, 0), -12.727926px -12.727926px 0 0 rgba(152, 128, 255, 0);
  animation: dot-spin 1.5s infinite linear;
}

@keyframes dot-spin {
  0%, 100% {
    box-shadow: 0 -18px 0 0 #9880ff, 12.727926px -12.727926px 0 0 #9880ff, 18px 0 0 0 #9880ff, 12.727926px 12.727926px 0 -5px rgba(152, 128, 255, 0), 0 18px 0 -5px rgba(152, 128, 255, 0), -12.727926px 12.727926px 0 -5px rgba(152, 128, 255, 0), -18px 0 0 -5px rgba(152, 128, 255, 0), -12.727926px -12.727926px 0 -5px rgba(152, 128, 255, 0);
  }
  12.5% {
    box-shadow: 0 -18px 0 -5px rgba(152, 128, 255, 0), 12.727926px -12.727926px 0 0 #9880ff, 18px 0 0 0 #9880ff, 12.727926px 12.727926px 0 0 #9880ff, 0 18px 0 -5px rgba(152, 128, 255, 0), -12.727926px 12.727926px 0 -5px rgba(152, 128, 255, 0), -18px 0 0 -5px rgba(152, 128, 255, 0), -12.727926px -12.727926px 0 -5px rgba(152, 128, 255, 0);
  }
  25% {
    box-shadow: 0 -18px 0 -5px rgba(152, 128, 255, 0), 12.727926px -12.727926px 0 -5px rgba(152, 128, 255, 0), 18px 0 0 0 #9880ff, 12.727926px 12.727926px 0 0 #9880ff, 0 18px 0 0 #9880ff, -12.727926px 12.727926px 0 -5px rgba(152, 128, 255, 0), -18px 0 0 -5px rgba(152, 128, 255, 0), -12.727926px -12.727926px 0 -5px rgba(152, 128, 255, 0);
  }
  37.5% {
    box-shadow: 0 -18px 0 -5px rgba(152, 128, 255, 0), 12.727926px -12.727926px 0 -5px rgba(152, 128, 255, 0), 18px 0 0 -5px rgba(152, 128, 255, 0), 12.727926px 12.727926px 0 0 #9880ff, 0 18px 0 0 #9880ff, -12.727926px 12.727926px 0 0 #9880ff, -18px 0 0 -5px rgba(152, 128, 255, 0), -12.727926px -12.727926px 0 -5px rgba(152, 128, 255, 0);
  }
  50% {
    box-shadow: 0 -18px 0 -5px rgba(152, 128, 255, 0), 12.727926px -12.727926px 0 -5px rgba(152, 128, 255, 0), 18px 0 0 -5px rgba(152, 128, 255, 0), 12.727926px 12.727926px 0 -5px rgba(152, 128, 255, 0), 0 18px 0 0 #9880ff, -12.727926px 12.727926px 0 0 #9880ff, -18px 0 0 0 #9880ff, -12.727926px -12.727926px 0 -5px rgba(152, 128, 255, 0);
  }
  62.5% {
    box-shadow: 0 -18px 0 -5px rgba(152, 128, 255, 0), 12.727926px -12.727926px 0 -5px rgba(152, 128, 255, 0), 18px 0 0 -5px rgba(152, 128, 255, 0), 12.727926px 12.727926px 0 -5px rgba(152, 128, 255, 0), 0 18px 0 -5px rgba(152, 128, 255, 0), -12.727926px 12.727926px 0 0 #9880ff, -18px 0 0 0 #9880ff, -12.727926px -12.727926px 0 0 #9880ff;
  }
  75% {
    box-shadow: 0 -18px 0 0 #9880ff, 12.727926px -12.727926px 0 -5px rgba(152, 128, 255, 0), 18px 0 0 -5px rgba(152, 128, 255, 0), 12.727926px 12.727926px 0 -5px rgba(152, 128, 255, 0), 0 18px 0 -5px rgba(152, 128, 255, 0), -12.727926px 12.727926px 0 -5px rgba(152, 128, 255, 0), -18px 0 0 0 #9880ff, -12.727926px -12.727926px 0 0 #9880ff;
  }
  87.5% {
    box-shadow: 0 -18px 0 0 #9880ff, 12.727926px -12.727926px 0 0 #9880ff, 18px 0 0 -5px rgba(152, 128, 255, 0), 12.727926px 12.727926px 0 -5px rgba(152, 128, 255, 0), 0 18px 0 -5px rgba(152, 128, 255, 0), -12.727926px 12.727926px 0 -5px rgba(152, 128, 255, 0), -18px 0 0 -5px rgba(152, 128, 255, 0), -12.727926px -12.727926px 0 0 #9880ff;
  }
}
