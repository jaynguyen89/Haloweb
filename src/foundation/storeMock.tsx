import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import React from 'react';
import { publicDataMock } from 'src/models/PublicData';

export const stageMock = {
    stages: [
        { name: 'LOADING_TEST' },
        {
            name: 'SHOW_FLASHER_SERVER_ERROR',
            type: 'error',
            message: 'This is server error test.',
        },
        {
            name: 'STAGE_FLASHER_TEST',
            message: '{{what}} Test',
            messageParams: { what: 'StageFlasher' },
            type: 'success',
        },
    ],
};

const mockStore = configureStore([]);
const storeMock = mockStore({
    languageStore: {
        siteLanguage: {
            selected: 'en',
            detected: 'en',
        },
    },
    publicDataStore: {
        publicData: publicDataMock,
    },
    stageStore: stageMock,
});

export const withStoreMock = (component: React.ReactElement) => {
    return (
        <Provider store={storeMock}>
            {component}
        </Provider>
    );
};
