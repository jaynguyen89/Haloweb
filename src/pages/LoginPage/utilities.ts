import { IAuthenticationData } from 'src/models/Authentication';
import { IRegionalizedPhoneNumber } from 'src/models/Profile';
import IPublicData from 'src/models/PublicData';
import {
    TFormDataState, TRangeOption, TSpecialOption,
    TValidatorOption,
    TValidatorOptionsMapFn,
    ValidatorNames,
} from 'src/utilities/data-validators/dataValidators';

export enum LoginFormFieldNames {
    EmailAddress = 'EmailAddress',
    AreaCode = 'AreaCode',
    PhoneNumber = 'PhoneNumber',
    Password = 'Password',
    Trusted = 'Trusted',
}

export const initialLoginFormDataState: TFormDataState<typeof LoginFormFieldNames> = {
    [LoginFormFieldNames.EmailAddress]: {
        value: undefined,
        caption: undefined,
    },
    [LoginFormFieldNames.AreaCode]: {
        value: undefined,
    },
    [LoginFormFieldNames.PhoneNumber]: {
        value: undefined,
        caption: undefined,
    },
    [LoginFormFieldNames.Password]: {
        value: undefined,
        caption: undefined,
    },
    [LoginFormFieldNames.Trusted]: { value: false },
};

export const loginFieldValidatorMap = {
    [LoginFormFieldNames.EmailAddress]: ValidatorNames.RangeValidator,
    [LoginFormFieldNames.AreaCode]: ValidatorNames.RangeValidator,
    [LoginFormFieldNames.PhoneNumber]: ValidatorNames.RangeValidator,
    [LoginFormFieldNames.Password]: ValidatorNames.RangeValidator,
};

export type TLoginFieldKey = Omit<typeof LoginFormFieldNames, 'Trusted'>;

export const loginValidatorOptionsMapFn: TValidatorOptionsMapFn<TLoginFieldKey> =
    (publicData?: IPublicData): TValidatorOption<TLoginFieldKey> =>
({
    [LoginFormFieldNames.EmailAddress]: {
        min: 6,
        max: 100,
    } as TRangeOption,
    [LoginFormFieldNames.AreaCode]: {
        among: publicData!.countries.map(country => country.telephoneCode),
    } as TRangeOption,
    [LoginFormFieldNames.PhoneNumber]: {
        equal: 9,
        numbersOnly: true,
    } as TRangeOption,
    [LoginFormFieldNames.Password]: {
        min: 8,
        max: 24,
    } as TRangeOption,
});

export const createLoginData = (formData: TFormDataState<typeof LoginFormFieldNames>): IAuthenticationData => ({
    emailAddress: formData[LoginFormFieldNames.EmailAddress].value as string | undefined,
    phoneNumber: formData[LoginFormFieldNames.PhoneNumber].value === undefined ? undefined : {
        phoneNumber: formData[LoginFormFieldNames.PhoneNumber].value,
        regionCode: formData[LoginFormFieldNames.AreaCode].value,
    } as IRegionalizedPhoneNumber,
    password: formData[LoginFormFieldNames.Password].value,
    isTrusted: formData[LoginFormFieldNames.Trusted].value,
    deviceInformation: undefined,
});
