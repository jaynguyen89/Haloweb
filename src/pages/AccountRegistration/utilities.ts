import { TRangeOption } from 'src/utilities/dataValidators';

export enum RegistrationValidatorFieldNames {
    EmailAddress = 'EmailAddress',
    AreaCode = 'AreaCode',
    PhoneNumber = 'PhoneNumber',
    Password = 'Password',
    PasswordConfirm = 'PasswordConfirm',
    Username = 'Username',
    Gender = 'Gender',
    GivenName = 'GivenName',
    MiddleName = 'MiddleName',
    FamilyName = 'FamilyName',
    FullName = 'FullName',
}

export const initialRegistrationFormDataState = {
    [RegistrationValidatorFieldNames.EmailAddress]: {
        value: undefined,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.AreaCode]: {
        value: undefined,
    },
    [RegistrationValidatorFieldNames.PhoneNumber]: {
        value: undefined,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.Password]: {
        value: undefined,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.PasswordConfirm]: {
        value: undefined,
    },
    [RegistrationValidatorFieldNames.Username]: {
        value: undefined,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.Gender]: {
        value: 4,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.GivenName]: {
        value: undefined,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.MiddleName]: {
        value: undefined,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.FamilyName]: {
        value: undefined,
        caption: undefined,
    },
    [RegistrationValidatorFieldNames.FullName]: {
        value: undefined,
        caption: undefined,
    },
};

export const validatorOptions = {
    [RegistrationValidatorFieldNames.EmailAddress]: {

    },
    [RegistrationValidatorFieldNames.AreaCode]: {

    },
    [RegistrationValidatorFieldNames.PhoneNumber]: {

    },
    [RegistrationValidatorFieldNames.Password]: {

    },
    [RegistrationValidatorFieldNames.Username]: {

    },
    [RegistrationValidatorFieldNames.Gender]: {

    },
    [RegistrationValidatorFieldNames.GivenName]: {

    },
    [RegistrationValidatorFieldNames.MiddleName]: {

    },
    [RegistrationValidatorFieldNames.FamilyName]: {

    },
    [RegistrationValidatorFieldNames.FullName]: {

    },
};
