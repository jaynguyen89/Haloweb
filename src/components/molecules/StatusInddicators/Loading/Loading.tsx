import React from 'react';
import { Box } from '@mui/material';
import 'src/components/molecules/StatusInddicators/Loading/loading.scss';

const Loading = () => {
    return (
        <Box sx={{ display: 'flex', alignItems: 'center' }}>
            <div data-title='dot-falling'>
                <div className='stage'>
                    <div className='dot-falling' />
                </div>
            </div>
        </Box>
    );
};

export default Loading;
