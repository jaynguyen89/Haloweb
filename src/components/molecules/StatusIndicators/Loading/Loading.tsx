import React from 'react';
import 'src/components/molecules/StatusIndicators/Loading/loading.scss';
import { useIsStageIncluded } from 'src/hooks/useStage';
import Stages from 'src/models/enums/stage';

export const variants = {
    falling: 'dot-falling',
    elastic: 'dot-elastic',
    pulsing: 'dot-pulse',
    flashing: 'dot-flashing',
    colliding: 'dot-collision',
    rotating: 'dot-carousel',
    jumping: 'dot-typing',
    circling: 'dot-windmill',
    floating: 'dot-floating',
};

type TLoading = {
    variant?: keyof typeof variants,
    stage: string,
};

const Loading = ({
    stage,
    variant = 'elastic',
}: TLoading) => {
    const visible = stage === Stages.SHOWCASE || useIsStageIncluded(stage);
    if (!visible) return null;

    return (
        <div data-title={variants[variant]}>
            <div className='stage'>
                <div className={variants[variant]} />
            </div>
        </div>
    );
};

export default Loading;
