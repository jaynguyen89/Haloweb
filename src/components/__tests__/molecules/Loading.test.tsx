import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import '@testing-library/jest-dom';
import Loading, { variants } from 'src/components/molecules/StatusIndicators/Loading/Loading';
import { render } from '@testing-library/react';
import { getByAttribute } from 'src/utilities/testUtilities';
import { withStoreMock } from 'src/foundation/storeMock';

configure({ adapter: new Adapter() });

describe('Loading.tsx', () => {
    it('renders successfully', () => {
        const wrapper = shallow(<Loading stage='showcase' />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should not be visible when stage is incorrect', () => {
        const wrapper = render(withStoreMock(<Loading stage='invisible' variant='falling' />));
        const loadingElement = getByAttribute('data-title')(wrapper.container, 'falling');
        expect(loadingElement).toBeNull();
    });

    it('should be visible when stage is correct', () => {
        const wrapper = render(withStoreMock(<Loading stage='LOADING_TEST' variant='falling' />));
        const loadingElement = getByAttribute('data-title')(wrapper.container, 'falling');
        expect(loadingElement).toBeDefined();
    });

    it('renders according to the specified props', () => {
        const kinds = Object.keys(variants);
        const wrappers = kinds.map(kind => render(withStoreMock(<Loading variant={kind} stage='LOADING_TEST' />)));

        kinds.forEach((kind, i) => {
            const loadingElement = getByAttribute('data-title')(wrappers[i].container, kind);
            expect(loadingElement).toBeDefined();

            const stageElement = wrappers[i].container.getElementsByClassName('stage')[0];
            expect(stageElement).toBeDefined();

            const animation = wrappers[i].container.getElementsByClassName(variants[kind])[0];
            expect(animation).toBeDefined();
        });
    });
});
