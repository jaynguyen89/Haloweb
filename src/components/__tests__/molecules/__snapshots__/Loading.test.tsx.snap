// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders successfully 1`] = `
<div
  data-title="dot-elastic"
>
  <div
    className="stage"
  >
    <div
      className="dot-elastic"
    />
  </div>
</div>
`;

exports[`Loading.tsx renders successfully 1`] = `
<div
  data-title="dot-elastic"
>
  <div
    className="stage"
  >
    <div
      className="dot-elastic"
    />
  </div>
</div>
`;
